src/app/components/home.component.ts

```diff
@@ -874,24 +874,21 @@ export class HomeComponent implements OnInit, AfterViewInit {
    * @param playerPath  full path to user's preferred video player
    * @param time        time in seconds
    */
-  public getVideoPlayerArgs(playerPath: string, time: number): string[] {
+  public getVideoPlayerArgs(playerPath: string, playerArgs: string, time: number): string[] {
     // if user doesn't want to open at timestamp, don't!
-    if (!this.settingsButtons['openAtTimestamp'].toggled) {
-      return [];
-    }
-
-    // else, figure out the correct command line flags
-    const argz: string[] = [];
+    const argz: string[] = playerArgs ? playerArgs.match(/[^\s"']+|"([^"]*)"/g) : [];
```

Comment:
Could you explain why there is a `.match()` happening?

I would have thought whatever the argument(s) the user adds in the input would just be added verbatim 🤔 

I'm not very familiar with how arguments get used with video players, but I thought it would be less puzzling to users to have the exact string just added as is 🤔 

Please let me know 🙂  👍  🤝

Reply:
The arguments is an array of string used to call [child_process.spawn()](https://nodejs.org/api/child_process.html#child_process_child_process_spawn_command_args_options). Substrings between spaces must be splitted into array elements to function. For example, if I want to spawn `player.exe /start /close "/jump 123"`, I need to call `spawn('player.exe', ['/start', '/close', '/jump 123'])` instead of `spawn('player.exe', ['/start /close "/jump=123"'])`, right? The regex is used to argument with spaces and quotes.

Sure, according to https://stackoverflow.com/questions/23487363/how-can-i-parse-a-string-into-appropriate-arguments-for-child-process-spawn, we can avoid these by using the `shell` option for `spawn()`, but I wanted to maintain the structure of the source code as much as possible. You guys probably had your reason to not use `shell`. If not, let me know or feel free to change yourself when merging.

Reply:
Thank you for the explanation! Please feel free to change `spawn()` to `shell` if that makes the code cleaner and/or has other benefits.

Please re-submit your PR -- it got closed by GitHub when I merged my branch and renamed `master` to `main` 😓

Reply:
ps - why not `.split(' ');` ? seems simpler if we're just splitting all the white spaces 🤔  or would this miss out on something?